homework/list.js

```diff
@@ -1,79 +1,126 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  /**
+  * Creates an instance of ListManager.
+  *
+  * @constructor
+  * @this {ListManager}
+  */
+  var ListManager = function () {
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper');
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    /**
+    * Fetch remote data to render list and
+    * bind 'click' event to initialize the ListManager.
+    *
+    * @this {ListManager}
+    */
+    start() {
+      this.fetchList()
+      .then((function(data) {
+        this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();
+      }).bind(this) )
+      .catch((function(){
+        //exception handle code here
```

Comment:
**Serious issue**: you can't just capture it and leave nothing. In fact, you need to **log AND re-throw** the error. To log it is to prevent the following chain make the error log silent, while to re-throw it is to make sure the same following chain will be interrupted if there is an error.